plugins/splunk-on-call/src/components/EntitySplunkOnCallCard.tsx

```diff
@@ -108,10 +151,38 @@ export const EntitySplunkOnCallCard = () => {
       {},
     );
     const teams = await api.getTeams();
-    const foundTeam = teams.find(teamValue => teamValue.name === team);
-    return { usersHashMap, foundTeam };
+    let foundTeams = [
+      teams.find(teamValue => teamValue.name === teamAnnotation),
+    ].filter(team => team !== undefined);
+
+    if (!foundTeams.length && routingKeyAnnotation) {
```

Comment:
☝️ Otherwise, if we haven't found associated teams -- and a `splunk.com/on-call-routing-key` annotation is used -- we proceed to find and return (as `foundTeams`) all teams associated with the routing key.